src/layer_builder.rs

```diff
@@ -34,28 +36,32 @@ where
     /// which will be used for rendering each chunk entity.
     pub fn new(
         commands: &mut Commands,
-        mut settings: LayerSettings,
+        settings: LayerSettings,
         map_id: impl MapId,
         layer_id: impl LayerId,
-    ) -> (Self, Entity) {
```

Comment:
removes tuple here.

1. entity is part of Self (LayerBuilder) anyway
2. entities for layers are managed internally anyway (new api doesn't even expose them). For other cases it can be extracted from LayerBuilder